source/opt/scalar_replacement_pass.h

```diff
@@ -39,7 +39,8 @@ class ScalarReplacementPass : public MemPass {
       : max_num_elements_(limit) {
     name_[0] = '\0';
     strcat(name_, "scalar-replacement=");
-    sprintf(&name_[strlen(name_)], "%d", max_num_elements_);
+    snprintf(&name_[strlen(name_)], sizeof(name_) - strlen(name_), "%d",
```

Comment:
Cache into local variable to avoid the redundant `strlen`

Reply:
Done